### PR TITLE
feat: remove splunk-appinspect dependency

### PR DESCRIPTION
**Pull request by arys-splunk**
This PR uses `addonfactory_splunk_conf_parser_lib` instead of `splunk_appinspect` for all `.conf` files parsers.

### CODE REVIEW

File: pytest_splunk_addon/standard_lib/addon_parser/fields.py [hunk 1]
```diff
@@ -51,6 +51,15 @@ def __init__(self, field_json=None):
     def __str__(self):
         return str(self.name)
 
+    def __eq__(self, other: "Field"):
+        return self.__dict__ == other.__dict__
+
+    def __lt__(self, other: "Field"):
```
Comment (hsekowski-splunk) [inline pytest_splunk_addon/standard_lib/addon_parser/fields.py:57]: Cannot see __gt__ in https://github.com/splunk/pytest-splunk-addon/blob/main/pytest_splunk_addon/standard_lib/addon_parser/fields.py
Probably worth to add, event if not needed now - just to have this part covered

**Review by hsekowski-splunk — DISMISSED** [review]
@arys-splunk ,
Beside comment regarding __gt__ , let me pay your attention to snyk that's failing (because of lack of some dependencies)

**Comment by arys-splunk** [comment]
`cim_field_parser` needs to be changed to respect new parsers.

https://github.com/splunk/pytest-splunk-addon/blob/10f0e496975fa49a3c0b78ead395e269cc45a495/pytest_splunk_addon/tools/cim_field_report.py#L501

File: pytest_splunk_addon/standard_lib/addon_parser/fields.py (hunk 1 above)
Comment (arys-splunk) [reply pytest_splunk_addon/standard_lib/addon_parser/fields.py:57]: Tests seem to be passing without this thing, if we need it in the future, I would like to add it in the future.

File: .github/workflows/build-test-release.yml [hunk 2]
```diff
@@ -121,7 +125,7 @@ jobs:
     strategy:
       fail-fast: false
       matrix:
-        splunk-version: ["8.0","8.1","8.2"]
+        splunk-version: ["8.1","8.2"]
```
Comment (hsekowski-splunk) [inline .github/workflows/build-test-release.yml:128]: My feeling it that should be 3 most recent versions

**Review by ricardom-splunk — APPROVED** [review]

File: .github/workflows/build-test-release.yml (hunk 2 above)
Comment (arys-splunk) [reply .github/workflows/build-test-release.yml:128]: Splunk 8.0 is not supported anymore: https://www.splunk.com/en_us/legal/splunk-software-support-policy.html

**Review by hsekowski-splunk — APPROVED** [review]

**Pull request merged**